frontend/lib/resolution/maybe-const.cpp

```diff
@@ -231,18 +231,18 @@ bool AdjustMaybeRefs::enter(const Call* ast, RV& rv) {
   // is it return intent overloading? resolve that
   if (candidates.numBest() > 1) {
     Access access = currentAccess();
+    auto kind = re.type().kind();
     bool ambiguity;
-    auto best = determineBestReturnIntentOverload(candidates, access, ambiguity);
+    auto best = determineBestReturnIntentOverload(candidates, access, kind, ambiguity);
```

Comment:
The initialization of ``kind`` here is a bit odd at first glance. I'd recommend initializing it to ``UNKNOWN``, passing it in to be potentially set, then checking for ``UNKNOWN`` again after in which case you set it to ``re.type().kind()``.

It's a bit convoluted, but I'm wary of passing in useful data to out-formals.